### DISCUSSION TITLE
Changing between mplayer & mpv in Ubuntu

### DISCUSSION THREAD

**Discussion by OldGeezer916** [smplayer-dev/smplayer#137] [Q&A]
When I first installed Smplayer in Ubuntu 18.04 the multimedia engine was mpv. I really liked everything about it except it would not display subtitles for videos that I had edited. I have a hearing problem, so I need subtitles. I experimented with the settings & nothing would fix it. I finally changed the engine to mplayer & that would show the subtitles. I had to  also take the driver off default to make it work. I'm using xv but other selections also worked.  There are a lot of things I would prefer with mpv. I would like to use it except for edited videos. Problem is it won't let me change back. I put the portable version on my Windows 8.1 machine & no problem changing between engines. It also would not show subtitles for my edited videos with mpv. I only use Smplayer in Windows to test videos. Ubuntu is my primary machine. My computer skills are only a little above average, so reply like I'm a newbie. :)

Here is the error log. Just looks like settings to me.

mpv -noquiet -slave -identify -nofs -sub-fuzziness 0 -autosync 100 -nodr -double -nomouseinput -input nodefault-bindings:conf=/dev/null -nokeepaspect -wid 128067173 -monitorpixelaspect 1 -osdlevel 0 -subfont-osd-scale 2 -progbar-align 80 -ass -embeddedfonts -ass-line-spacing 0 -ass-font-scale 1 -noflip-hebrew -ass-force-style PlayResX=512,PlayResY=320,Name=Default,Fontname=Arial [TMC ],Fontsize=18,PrimaryColour=&H0000af00,BackColour=&H00000000,OutlineColour=&H00000000,Bold=1,Italic=0,Alignment=2,BorderStyle=1,Outline=0.3,Shadow=1,MarginL=1,MarginR=1,MarginV=5 -subcp ISO-8859-1 -subpos 100 -contrast 13 -brightness 19 -hue 2 -saturation 15 -vf-add eq2,hue -volume 56 -cache 6144 -ss 1065 -vf-add pp -autoq 6 -vf-add expand=aspect=1.77778,harddup -noslices -channels 2 -af-add scaletempo -af-add equalizer=-12:-9.2:-3.5:0:0:0:0:0:0:0 -softvol -softvol-max 150 /media/john/512/Watch/Curb Your Enthusiasm 1080 AMZN WEB HEVC/Curb.Your.Enthusiasm.S06E07.The.TiVo.Guy.1080p.AMZN.WEB-DL.DD+2.0.H.264-AJP69 (Encoded).mkv

**Comment by smplayer-dev** [comment]
There isn't any error in the log, which may mean that mpv failed to run. If you open an terminal and type `mpv` do you get any error?

By the way it's very strange that mpv couldn't display subtitles. Maybe if you send me a sample subtitle file I could try to figure out why the subtitles are not working.

**Reply by OldGeezer916** [reply]
Thanks for your reply. Part of my problem is now fixed.

The subtitles are a track in the video, not a srt or sub file. I use Xillisoft Video Converter Ultimate to clip the ads off the beginning & end of some files. Files that I crop or convert to HEVC with VidCoder don't give me any problem with subs. 

Thanks to you telling me to type mpv in the terminal. I found out why I could not change to that engine. It told me to enter "sudo apt install mpv". Somehow mpv must have been uninstalled by one of the many things I tried. That put it back. Now I can use it most of the time.
 
As I said before, I have the same problem with the subs in Windows portable. In mpv, under the subtitle menu it shows the track selected. In the subtitle menu the visibility is checked. When the video starts it shows the track in the on screen display. Just can't get them visible unless I switch to mplayer with my clipped videos. Then they are fine. Except for not showing the subs from my clipped files mpv is the much better engine.

So glad I found Smplayer. Except for a couple things it is SO much better than VLC. Every other player I tried was worse than that crappy Windows Media!

**Reply by smplayer-dev** [reply]
Could you send the mpv log (menu View) after selecting the subtitle track?

**Reply by OldGeezer916** [reply]
This is a clipped file with mpv selected. Subs are selected but do not show. Switch to mplayer & they work fine. Would it help to send you the log of the same video with mplayer engine too?

[log.txt](https://github.com/smplayer-dev/smplayer/files/6770070/log.txt)

**Reply by smplayer-dev** [reply]
Yes, maybe it can help the mplayer log as well.

Unfortunately the mpv log doesn't say why it doesn't show the subtitles. Although I noticed you're using an old version of mpv (0.27.2, current version is 0.33.1), maybe you can try to update it.

**Reply by OldGeezer916** [reply]
Same file. Mplayer won't work in Ubuntu unless you change the driver from default. Several options work, some don't. I'm using xv. With Windows portable you can use default for both. In Ubuntu mpv is kind of squirrely. When you pause & select anything else the window goes black. It comes back on as soon as you hit play. Does not do this in Windows.

 As far as updates that is entirely up to the Ubuntu team. When I had 14.04 they never updated VLC from version 2.2.4 in a couple years I used it even though 3.x was out. After I upgraded to 18.04 my VLC version is 3.0.8. No updates even though VLC 3.0.16 is out. I don't expect to see an update until I upgrade to 20.04.

[log mplqayer.txt](https://github.com/smplayer-dev/smplayer/files/6770438/log.mplqayer.txt)

**Reply by smplayer-dev** [reply]
Unfortunately I can't see any useful information in the mplayer log.
Can you provide a small mkv sample video with this problem?

**Reply by OldGeezer916** [reply]
It won't let me attach a mkv file so I changed the extension. Just rename it back to mkv. Checked it out. Subs show up in mplayer, not in mpv.

https://user-images.githubusercontent.com/74526703/124628093-c25e4600-de45-11eb-80e2-1a2e56d69cec.mp4

**Reply by OldGeezer916** [reply]
Doesn't look like that will work.

**Reply by smplayer-dev** [reply]
I can reproduce the problem with your video. I'll take a look at it and see if I can fix it.

**Comment by smplayer-dev** [comment]
There are several weird things. I extracted the subtitles from the mkv:

```
﻿1
00:00:00,744 --> 00:00:00,744
All of the water
from the dam caused more damage

2
00:00:02,789 --> 00:00:02,789
than just flooding
and mudslides.

3
00:00:04,400 --> 00:00:04,400
It quickly eroded the bottom
of the hill, and then--

4
00:00:06,489 --> 00:00:06,489
Brought the whole hill down.

5
00:00:07,838 --> 00:00:07,838
Most of the damage is
concentrated east of the dam.

6
00:00:10,362 --> 00:00:10,362
Exposed gas lines,
unstable structures, and mud.

```

The timestamps seems to be wrong, the ending timestamps are the same as the beginning timestamps. Anyway mplayer displays the subtitles, using -dumpsrtsub it seems mplayer fixes the ending timestamps:

```
1
00:00:00,740 --> 00:00:02,770
All of the water
from the dam caused more damage

2
00:00:02,780 --> 00:00:04,390
than just flooding
and mudslides.

3
00:00:04,400 --> 00:00:06,470
It quickly eroded the bottom
of the hill, and then--

4
00:00:06,480 --> 00:00:07,820
Brought the whole hill down.

5
00:00:07,830 --> 00:00:10,350
Most of the damage is
concentrated east of the dam.

6
00:00:10,360 --> 00:00:16,360
Exposed gas lines,
unstable structures, and mud.

```

But another weird thing, if I open the mkv with MKVToolNix GUI and then save it as a new video, mpv now displays the subtitles!

So I don't know, but I think the problem is caused by the software you're using to create the mkv files.

**Reply by OldGeezer916** [reply]
I used MkvToolNix to mux a new file & still no good. The subs do not show on mpv. Tried playing the new file in both Windows & Ubuntu. MkvToolNix is great at fixing a lot of problems in a video file, but it didn't work for me on this. I downloaded the latest version. I have been using Xillisoft convertor for years. I have found no other  program that can so precisely clip & join files.

**Reply by smplayer-dev** [reply]
Where do you get the subtitles? Have you checked if the format is correct?

**Reply by OldGeezer916** [reply]
It is the sub track in the file. I used the same video as the 10 second clip I sent you.

**Reply by smplayer-dev** [reply]
I think the problem is in the subtitles. I attach the same video but with the subtitles taken from addic7ed. Now mpv displays the subtitles.

[video-subs-fixed.zip](https://github.com/smplayer-dev/smplayer/files/6777846/video-subs-fixed.zip)

**Reply by OldGeezer916** [reply]
The subs are working in that file with mpv. I have used MkvToolNix before to just remove the sub tracks that weren't working & downloaded from opensubtitles. I just used separate srt files to watch them, but I could have muxed them into the file. That's a lot of trouble to go to. Mplayer is not near as good as mpv. It's kind of buggy in Ubuntu, but I may just have to use it with the edited files.
